src/legendre_fet.cpp

```diff
@@ -196,40 +195,255 @@ void LegendreFET::score_source(const BankedParticle& p) {
+      switch (axes_[it_axis]) {
+        case LegendreFET::Axis::X: {
+          // get the x-min of the position bin and calculate the scaled-x
+          const double xmin_ = cartesian_filter_->x_min(position_index);
+          const double scaled_x = 2. * (tktr.r().x() - xmin_) * inv_dx_ - 1.;
+          // get the value of each legendre polynomial for calculated scaled-x
+          const std::vector<double> lgendre_value =
+              legendre_polynomial_.evaluate_legendres(scaled_x);
+          // do the loop over each order and calculate the fet in x-direction
+          double fet_value_x = 0.;
+          for (std::size_t order = 0; order <= fet_order_; order++) {
+            indices[FET_index] = order;
+            fet_value_x += (2. * static_cast<double>(order) + 1.) *
+                           tally_avg_.element(indices.begin(), indices.end()) *
+                           lgendre_value[order];
+          }
+          tally_value *= fet_value_x;
+        } break;
+
+        case LegendreFET::Axis::Y: {
+          // get the y-min of the position bin and calculate the scaled-y
+          const double ymin_ = cartesian_filter_->y_min(position_index);
+          const double scaled_y = 2. * (tktr.r().y() - ymin_) * inv_dy_ - 1.;
+          // get the value of each legendre polynomial for calculated scaled-y
+          const std::vector<double> lgendre_value =
+              legendre_polynomial_.evaluate_legendres(scaled_y);
+          // do the loop over each order and calculate the fet in y-direction
+          double fet_value_y = 0.;
+          for (std::size_t order = 0; order <= fet_order_; order++) {
+            indices[FET_index] = order;
+            fet_value_y += (2. * static_cast<double>(order) + 1.) *
+                           tally_avg_.element(indices.begin(), indices.end()) *
+                           lgendre_value[order];
+          }
+          tally_value *= fet_value_y;
+        } break;
+
+        case LegendreFET::Axis::Z: {
+          // get the z-min of the position bin and calculate the scaled-z
+          const double zmin_ = cartesian_filter_->z_min(position_index);
+          const double scaled_z = 2. * (tktr.r().z() - zmin_) * inv_dz_ - 1.;
+          // get the value of each legendre polynomial for calculated scaled-z
+          const std::vector<double> lgendre_value =
+              legendre_polynomial_.evaluate_legendres(scaled_z);
+          // do the loop over each order and calculate the fet in z-direction
+          double fet_value_z = 0;
+          for (std::size_t order = 0; order <= fet_order_; order++) {
+            indices[FET_index] = order;
+            fet_value_z += (2. * static_cast<double>(order) + 1.) *
+                           tally_avg_.element(indices.begin(), indices.end()) *
+                           lgendre_value[order];
+          }
+          tally_value *= fet_value_z;
+        }
+        default:
+          break;
```

Comment:
Because we are using an enum class here, I don't think we need the default. If we do eventually need the default though, I think we should throw a fatal_error here, as we should never get here.

Reply:
Should I throw the `fatal_error` in other functions like `score_collision` as well?

Reply:
I have removed the default from all the other places as well.